tests/integration/helpers/cluster.py

```diff
@@ -1434,6 +1434,7 @@ def start(self, destroy_dirs=True):
                 for dir in self.zookeeper_dirs_to_create:
                     os.makedirs(dir)
                 run_and_check(self.base_zookeeper_cmd + common_opts, env=self.env)
+                self.up_called = True
```

Comment:
Does it mean that in some cases docker-compose was not stopped? Was it the reason we run out of some resources?

It is good to refactor - maybe have a separate class working with docker stuff (so that this flag is set in one place).
Maybe later.

Reply:
This flag introduced long ago and helped to have correct logs in cases when test script is totally incorrect and it did not call `docker-compose up`. Before this trivial hack we tried to call `docker-compose logs` and it failed. Currently I met case when we called up for dependencies and not for clickhouse-server itself and didn't try to copy logs from those docker containers. Added this trivial way to call docker-compose logs in case dependencies failed. 
FYI we call docker-compose up two times for dependencies and for clickhouse after we checked that they are online